pkg/net/neighborhood.go

```diff
@@ -243,14 +225,20 @@ func (vtx *vertex) Load() edgeMap {
 	return nil
 }
 
+func (vtx *vertex) Len() int {
+	return len(vtx.Load())
+}
+
 func (vtx *vertex) Store(es edgeMap) { vtx.value.Store(es) }
 
 type edge struct {
-	cq chan struct{}
-	s  network.Stream
+	cq  chan struct{}
+	s   network.Stream
+	rec *peer.PeerRecord
 }
 
 func (e edge) ID() peer.ID              { return e.s.Conn().RemotePeer() }
```

Comment:
Okay, but why did you write with capitals? `ID()` is only used within the same package right?

Reply:
It's an informal/personal convention I use to help me keep track of which fields are accessed by other types.  It basically sends a signal that says "don't touch this" (in the case of `cq` and `s`).  It's kind of like the `_foo` convention in Python, if that makes sense.

Reply:
I think we should follow the same convention on naming throughout the project. So, we should rename `rec` and `recs` to `peer` and `peers`. However, it may conflict with the `peer` package we import from libp2p. What do you think?

Reply:
Although I'm generally of a mind to keep things consistent, I think I prefer to leave this as-is.

My reasoning is that as you said, `peer` would collide with the imported package.  And while `rec` is definitely less informative (it's a record of _what_?), in practice it only appears in function signatures, where it's type is readily visible.  The same is not true of a struct field.  When you see `e.Rec`, you can't just glance up at the function signature -- you need to navigate to the type definition, which may be in another file.  But `e.Peer` is much clearer.

Reply:
Oh, on the other hand, we could use `remote` and `view` instead of `peer` and `peers`.

This would give us `e.Remote` in the case of `edge`, which is perfectly understandable (even more so with `e.Remote.ID` or `e.Remote.Addrs`).

Your call 🙂